dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/AdvancedSearchPage.java

```diff
@@ -260,19 +277,31 @@ private boolean isDateField(DatasetFieldType datasetFieldType) {
                                                        BundleUtil.getStringFromBundle("description"),
                                                        BundleUtil.getStringFromBundle("advanced.search.dataverses.description.tip")));
 
-        CheckboxSearchField checkboxSearchField = new CheckboxSearchField(SearchFields.DATAVERSE_SUBJECT,
-                                                                          BundleUtil.getStringFromBundle("subject"),
-                                                                          BundleUtil.getStringFromBundle("advanced.search.dataverses.subject.tip"));
-
         DatasetFieldType subjectType = datasetFieldService.findByName(DatasetFieldConstant.subject);
 
-        for (ControlledVocabularyValue vocabValue : subjectType.getControlledVocabularyValues()) {
-            checkboxSearchField.getCheckboxLabelAndValue().add(Tuple.of(vocabValue.getLocaleStrValue(),
-                                                                        vocabValue.getStrValue()));
+        if(subjectType.isThisOrParentAllowsMultipleValues()) {
```

Comment:
Actually it shouldn't matter if subject field type can have multiple subjects or not (And always render as CheckboxSearchField).
Reason behind that is how search is constructed. When selecting this value we want to search for dataverses that contains datasets with that subject. As dataverse can have many datasets then for single dataverse we always can have multiple subjects regardless of allowMulti for subject field in dataset